docs/data/data-grid/editing/editing.md

```diff
@@ -4,266 +4,411 @@ title: Data Grid - Editing
 
-{{"demo": "RowEditControlGrid.js", "bg": "inline", "defaultCodeOpen": false}}
+```tsx
+apiRef.current.setEditCellValue({ id, field, value: newValue, debounceMs: 200 });
+```
 
-### Conditional validation [<span class="plan-pro"></span>](https://mui.com/store/items/material-ui-pro/)
+Since now a new render only occurs after some time has passed, the `value` prop will not be updated on each `apiRef.current.setEditCellValue` call.
+To avoid frozen UIs, the edit component can keep the current value in an internal state and sync it once `value` changes.
+To do that apply the following modifications to your edit compoment.
+
+```diff
+ function CustomEditComponent(props: GridRenderEditCellParams) {
+-  const { id, value, field } = props;
++  const { id, value: valueProp, field } = props;
++  const [value, setValue] = React.useState(valueProp);
+   const apiRef = useGridApiContext();
+
+   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+     const newValue = event.target.value; // The new value entered by the user
+-    apiRef.current.setEditCellValue({ id, field, value: newValue });
++    apiRef.current.setEditCellValue({ id, field, value: newValue, debounceMs: 200 });
++    setValue(newValue);
+   };
+
++  React.useEffect(() => {
++    setValue(valueProp);
++  }, [valueProp]);
++
+   return <input type="text" value={value} onChange={handleChange}>;
+}
+```
 
-Having all cells of a row in edit mode allows validating a field based on the value of another one.
-To accomplish that, start by adding the `preProcessEditCellProps` as explained in the [cell editing](#client-side-validation).
-When the callback is called, use the API to check the value of the other field and decide if the current value is valid or not.
-Return a new object contaning `params.props` and the `error` attribute with the validation status.
-Once at the least one field has the `error` attribute equals to true no new value will be committed.
+### With auto-stop
 
-**Note:** For server-side validation, the same [approach](#server-side-validation) from the cell editing can be used. The data grid will wait for all promises to resolve before commiting.
+An edit component has an "auto-stop" behavior when it stops the edit mode as soon as the value is changed.
+To picture better, imagine an edit component with a combo, created following the steps above.
+By default, it would require two clicks to change the value of the cell: one click to select a new value and another click outside to save.
+This second click can be avoided if the first click also stops the edit mode.
+To create an edit component with auto-stop, call `apiRef.current.stopCellEditMode` after setting the new value.
+Since `apiRef.current.setEditCellValue` may do additional processing, it is necessary to first wait for it to resolve before stopping the edit mode.
```

Comment:
```suggestion
Since `apiRef.current.setEditCellValue` may do additional processing, you must wait for it to resolve before stopping the edit mode.
```